services/ranking_service.py

```
@@ -0,0 +1,190 @@
+        # Get embeddings for query and passages
+        start_time = time.time()
+        try:
+            # Embed query
+            query_embedding = embed_texts([query])[0]
+            
+            # Embed passages
+            passage_embeddings = embed_texts(passages)
+            
+            # Compute similarity scores
+            similarity_scores = self._compute_similarity(query_embedding, passage_embeddings)
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)